plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeTransactionLogEntryIterator.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.transactionlog;
+
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.plugin.deltalake.transactionlog.checkpoint.LastCheckpoint;
+import io.trino.spi.TrinoException;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.plugin.deltalake.DeltaLakeErrorCode.DELTA_LAKE_INVALID_SCHEMA;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogParser.readLastCheckpoint;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogDir;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogJsonEntryPath;
+import static io.trino.plugin.deltalake.transactionlog.checkpoint.TransactionLogTail.getEntriesFromJson;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakeTransactionLogEntryIterator
+        implements Iterator<DeltaLakeTransactionLogEntry>
+{
+    private final TrinoFileSystem fileSystem;
+    private final Path transactionLogDir;
+    private long currentVersion;
+    private Optional<Iterator<DeltaLakeTransactionLogEntry>> currentEntries = Optional.empty();
```

Comment:
IMO, it would be preferable to utilize `CommitInfoEntry commitInfo;` instead. I fail to see the necessity of an `Iterator` when our goal is to load solely the `commitInfo` entry from the transaction log.

Reply:
This will reduce the workload on the `hasNext` implementation.

Reply:
I opted (at least for now) for all the entries of the transaction log file `List<DeltaLakeTransactionLogEntry>` - this makes the handling indeed easier.

Doing the iterator over `DeltaLakeTransactionLogEntry` items is over-engineering.